src/test/java/uk/gov/justice/probation/courtcaseservice/service/OffenderMatchServiceTest.java

```diff
@@ -50,212 +54,290 @@
     @Mock
     private GroupedOffenderMatchRepository offenderMatchRepository;
     @Mock
-    private CourtCaseEntity courtCaseEntity;
-    @Mock
     private GroupedOffenderMatchesEntity groupedOffenderMatchesEntity;
     @Mock
     private GroupedOffenderMatchesRequest groupedOffenderMatchesRequest;
+    @Mock
+    private CourtCaseEntity courtCaseEntity;
 
     private OffenderMatchService service;
 
     @BeforeEach
-    public void setUp() {
+    void setUp() {
         when(offenderRestClientFactory.build()).thenReturn(offenderRestClient);
         service = new OffenderMatchService(courtCaseService, offenderMatchRepository, offenderRestClientFactory);
     }
 
-    @Test
-    void givenValidRequest_whenCreateGroupedMatchesCalled_thenCreateAndReturnMatch() {
-        when(offenderMatchRepository.findByCourtCodeAndCaseNo(COURT_CODE, CASE_NO)).thenReturn(Optional.empty());
-        when(courtCaseService.getCaseByCaseNumber(COURT_CODE, CASE_NO)).thenReturn(courtCaseEntity);
-        when(offenderMatchRepository.save(any(GroupedOffenderMatchesEntity.class))).thenReturn(groupedOffenderMatchesEntity);
+    @Nested
+    class CreateOrUpdateByCourtAndCaseNo {
```

Comment:
I've got to say I'm not sure I'm a fan of nested test classes, I find it difficult to track the different methods and variables in scope and have found myself inadvertently editing the wrong data/methods etc as a result of multiple nested tests shadowing each other.  Maybe it's just lack of familiarity on my part, not suggesting you change it, just something I've been mulling 🤔